php/admin/class-admin.php

```diff
@@ -93,11 +93,11 @@ public function enqueue_scripts() {
 
 	/**
 	 * Redirect to the Settings screen if the license is being saved.
+	 *
+	 * This runs before get_current_screen() returns a value,
 	 */
 	public function maybe_settings_redirect() {
-		$page = filter_input( INPUT_GET, 'page', FILTER_SANITIZE_STRING );
-
-		if ( 'block-lab-pro' === $page ) {
+		if ( is_admin() && isset( $_GET['page'] ) && 'block-lab-pro' === $_GET['page'] ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
```

Comment:
This is changed because I couldn't figure out how to mock a `filter_input()` value in unit tests. Maybe there's a way.